src/vk/vk-posts/vkpostsyncadaptor.cpp

```diff
@@ -0,0 +1,472 @@
+    if (!photosObject.isEmpty()) {
+        QJsonArray photoArray = photosObject.value(QStringLiteral("items")).toArray();
+        Q_FOREACH (const QJsonValue &photoValue, photoArray) {
+            QJsonObject photo = photoValue.toObject();
+            QString src = photo.value(m_optimalImageSize).toString();
+            if (!src.isEmpty()) {
+                photoOwnerId = (int)photo.value(QStringLiteral("owner_id")).toDouble();
+                photoAlbumId = (int)photo.value(QStringLiteral("album_id")).toDouble();
+                images.append(qMakePair(src, SocialPostImage::Photo));
+                hasValidContent = true;
+            }
+        }
+    }
+
+    QDateTime createdTime = VKDataTypeSyncAdaptor::parseVKDateTime(post.value(QStringLiteral("date")));
+    QString posterName, posterIcon;
+    int fromId = newPost.fromId;
+    if (newPost.fromId < 0) {
+        // it was posted by a group
+        const GroupProfile &group(VKDataTypeSyncAdaptor::findGroupProfile(groupProfiles, newPost.fromId));
+        posterName = group.name;
+        posterIcon = group.icon;
+        fromId = -fromId;
+    } else {
+        // it was posted by a user
+        const UserProfile &user(VKDataTypeSyncAdaptor::findUserProfile(userProfiles, newPost.fromId));
+        posterName = user.name();
+        posterIcon = user.icon;
+    }
+
+    // VK post indentifier is just index number and not globally unique. To make
+    // it unique we combine it with fromId
+    QString identifier = QString::number(fromId) + QStringLiteral("_") +
+                         (post.contains(QStringLiteral("id"))
+                            ? QString::number((int)post.value(QStringLiteral("id")).toDouble())
+                            : QString::number((int)post.value(QStringLiteral("post_id")).toDouble()));
+
+    newPost.link = QStringLiteral("https://m.vk.com/album")
+              + QString::number(photoOwnerId)
+              + QStringLiteral("_")
+              + QString::number(photoAlbumId);
+
+    if (hasValidContent) {
+        m_db.addVKPost(identifier, createdTime, QString(), newPost, images, posterName, posterIcon, accountId);
+    } else {
+        SOCIALD_LOG_DEBUG("VK photo post without valid content, skipping");
+    }
+}
+
+void VKPostSyncAdaptor::determineOptimalImageSize()
+{
+    int width = 0, height = 0;
+    const int defaultValue = 0;
+    MGConfItem widthConf("/lipstick/screen/primary/width");
+    if (widthConf.value(defaultValue).toInt() != defaultValue) {
+        width = widthConf.value(defaultValue).toInt();
+    }
+    MGConfItem heightConf("/lipstick/screen/primary/height");
+    if (heightConf.value(defaultValue).toInt() != defaultValue) {
+        height = heightConf.value(defaultValue).toInt();
+    }
+
+    // we want to use the largest of these dimensions as the "optimal"
+    int maxDimension = qMax(width, height);
+    if (maxDimension >= 2048) {
+        m_optimalImageSize = "photo_1280";
```

Comment:
That was the highgest the spec mentioned, but there actually is higher one in the server reply.